R/util.R

```diff
@@ -79,3 +84,30 @@ unpack_tarball <- function(pkg_tar, temp_file_name = "SCORECARD_"){
   return(pkg_source_path)
 }
 
+
+#' Warn that a feature is deprecated
+#'
+#' @param version mpn.scorecard version when something became deprecated
+#' @param what A string describing what is deprecated.
+#' @param details Can either be a single string or a character vector, which
+#'   will be converted to a bulleted list.
+#'
+#' @details
+#' This function is different from lifecycle::deprecate_warn, as it does not
+#' refer to the function that called it, nor care about the environment it was
+#' called in. It simply triggers an inforamtive warning if the current
+#' `mpn.scorecard` version is ahead of the `version` supplied.
+#'
+#' @noRd
+deprecate_warning <- function(version, what, details = NULL){
+  # mpn.scorecard version
+  mpn_scorecard_ver <- as.character(utils::packageVersion("mpn.scorecard"))
```

Comment:
@barrettk @seth127 I was just taking a glance at the warning now to know what to expect for the score-r-pkg script adjustments, and I'm confused by this `as.character`.

Why coerce this to a character?  Doesn't that compare the wrong thing in the condition?

Example of a bad case:

```R
> as.character(numeric_version("2.0.0")) > as.character(numeric_version("11.0.0"))
[1] TRUE
```

Reply:
For completeness, the good variant of that, with `numeric_version` coming from `packageVersion` and a string coming from the caller, would be

```R
> numeric_version("2.0.0") > "11.0.0"
[1] FALSE
```